public/app/features/provisioning/Wizard/JobStep.tsx

```diff
@@ -0,0 +1,111 @@
+import { skipToken } from '@reduxjs/toolkit/query';
+import { ReactNode, useEffect, useRef, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { Box, Spinner, Stack, Text } from '@grafana/ui';
+
+import { JobStatus } from '../JobStatus';
+import { useListJobQuery } from '../api';
+import { StepStatus, useStepStatus } from '../hooks/useStepStatus';
+
+import { WizardFormData } from './types';
+
+interface JobStepProps {
+  onStepUpdate: (status: StepStatus, error?: string) => void;
+  description: ReactNode;
+  startJob: (repositoryName: string) => Promise<{ metadata?: { name?: string } }>;
+  children?: ReactNode;
+}
+
+export type { JobStepProps };
+
+export function JobStep({ onStepUpdate, description, startJob, children }: JobStepProps) {
+  const hasInitialized = useRef(false);
+  const { watch } = useFormContext<WizardFormData>();
+  const repositoryName = watch('repositoryName');
+  const stepStatus = useStepStatus({ onStepUpdate });
+  const [jobName, setJobName] = useState<string>();
+
+  // Query the job status if we have a job name
+  const jobQuery = useListJobQuery(jobName ? { watch: true, fieldSelector: `metadata.name=${jobName}` } : skipToken);
+
+  useEffect(() => {
```

Comment:
Look at using `useAsync` hook to avoid manually tracking the mounted/initialised states.